ruby/ql/lib/codeql/ruby/security/MassAssignmentCustomizations.qll

```diff
@@ -52,4 +53,45 @@ module MassAssignment {
       result.(DataFlow::PostUpdateNode).getPreUpdateNode() = this.getParamsArgument()
     }
   }
+
+  /** Holds if `h` is an empty hash or contains an empty hash at one if its (possibly nested) values. */
+  private predicate hasEmptyHash(Expr e) {
```

Comment:
It's better to use CFG nodes instead of AST nodes here. I.e., use `HashLiteralCfgNode` instead of `HashLiteral`, etc.